content/client-lib-development-guide/features.textile

```diff
@@ -76,6 +76,7 @@ h3(#restclient). RestClient
 ** @(RSC7a)@ The header @X-Ably-Version: 1.1@ must be included in all REST requests to the Ably endpoint
 ** @(RSC7b)@ The header @X-Ably-Lib: [lib][.optional variant]?-[version]@ should be included in all REST requests to the Ably endpoint where @[lib]@ is the name of the library such as @js@ for @ably-js@, @[.optional variant]@ is an optional library variant, such as @laravel@ for the @php@ library, which is always delimited with a period such as @php.laravel@, and where @[version]@ is the full client library version using "Semver":http://semver.org/ such as @1.0.2@. For example, the 1.0.0 version of the Javascript library would use the header @X-Ably-Lib: js-1.0.0@
 *** @(RSC7b1)@ When it is not possible to send the @X-Ably-Lib@ header, such as for @JSONP@ requests, the library version should be sent as a query param such as @lib=js-1.0.0@
+** @(RSC7c)@ If the @addRequestIds@ client option is enabled, every REST request to Ably should include in a @request_id@ query string parameter a random string obtained by url-safe base64-encoding a sequence of at least 9 bytes obtained from a source of randomness. This request ID must remain the same if a request is retried to a fallback host per @RSC15@. If the request fails, the request ID must be included in the @ErrorInfo@ returned to the user, and printed as part of any corresponding error log
```

Comment:
Should we not output this id if a request fails as well, even if just the first time, so that customers can see these issues in their logs if they are interested. They would be warn only though.  I think that is what we do with Ruby.

Reply:
Isn't that achieved by `and printed as part of any corresponding error log`?

Reply:
Well it’s a bit ambiguous because an error is only returned if all attempts fail, so one would naturally assume the .. and .. part refers to the corresponding error log to the error returned, which does not happen if the first attempt fails

Reply:
fair point, I'll change to `Any log messages associated with the request should include the request ID. If the request fails, the request ID must be included in the @ErrorInfo@ returned to the user.`